src/game.cpp

```diff
@@ -851,6 +852,77 @@ bool game::start_game()
     return true;
 }
 
+static bool compare_distance( const tripoint &goal_a, const tripoint &goal_b )
+{
+    tripoint origin = g->u.global_omt_location();
+    return rl_dist( origin, goal_a ) < rl_dist( origin, goal_b );
+}
+
+void game::place_starting_vehicle( cata::optional<vproto_id> starting_vehicle )
```

Comment:
I'd suggest accepting `starting_vehicle` by constant reference and pull the condition up to the calling code.